apps/api/src/app/rate-limiting/usecases/get-api-rate-limit-service-maximum-config/get-api-rate-limit-service-maximum-config.usecase.ts

```diff
@@ -0,0 +1,86 @@
+import { Injectable, OnModuleInit } from '@nestjs/common';
+import {
+  ApiRateLimitCategoryEnum,
+  ApiRateLimitServiceMaximumEnvVarFormat,
+  ApiServiceLevelEnum,
+  DEFAULT_API_RATE_LIMIT_SERVICE_MAXIMUM_CONFIG,
+  IApiRateLimitServiceMaximum,
+} from '@novu/shared';
+import { createHash } from 'crypto';
+import {
+  buildMaximumApiRateLimitKey,
+  buildServiceConfigApiRateLimitMaximumKey,
+  CacheService,
+  InvalidateCacheService,
+} from '@novu/application-generic';
+
+@Injectable()
+export class GetApiRateLimitServiceMaximumConfig implements OnModuleInit {
+  public default: IApiRateLimitServiceMaximum = DEFAULT_API_RATE_LIMIT_SERVICE_MAXIMUM_CONFIG;
+
+  constructor(private invalidateCache: InvalidateCacheService, private cacheService: CacheService) {}
+
+  async onModuleInit() {
+    await this.loadDefault();
+  }
+
+  public async loadDefault(): Promise<void> {
+    const newDefault = this.createDefault();
+    this.default = newDefault;
+
+    if (!this.cacheService.cacheEnabled()) {
+      return;
+    }
+
+    const cacheKey = buildServiceConfigApiRateLimitMaximumKey();
+    const previousHash = await this.cacheService.get(cacheKey);
+    const newHash = this.getConfigHash(newDefault);
+
+    if (previousHash !== newHash) {
+      await this.cacheService.set(cacheKey, newHash);
+
+      this.invalidateCache.invalidateQuery({
+        key: buildMaximumApiRateLimitKey({
+          _environmentId: '*',
+          apiRateLimitCategory: '*',
```

Comment:
what is the motivation behind the `*` character?

Reply:
The intention here is to cater to the following scenario:
1. A Novu platform administrator makes changes to the default API rate limit service config environment variables. All [cached maximum environment limits](https://github.com/novuhq/novu/blob/3c1fd8a30889097da4840dd19cd7975dffc67453/apps/api/src/app/rate-limiting/usecases/get-api-rate-limit-maximum/get-api-rate-limit-maximum.usecase.ts#L26) for all Organizations, for all Environments need to be invalidated. This cached entity supports fast lookups for an Environment's max limit.
2. When a change is detected (via the modified hash), invalidate all cached entities.
3. Update the hash of the service config

This pattern could be used to support other platform service configuration variables that require cache invalidation. This will typically be the case when entities are cached for performance reasons, but still depend on defaults.

Reply:
Amazing!